lib/ability.rb

```diff
@@ -28,7 +28,7 @@ def initialize(user, resource, controller)
         can :manage, [AccountPriceGroupMember, UserPriceGroupMember]
       end
 
-      if resource.global? && (user.user_roles.map(&:role) & UserRole.facility_roles).any?
+      if resource.global? && (user.user_roles.map(&:role) & UserRole.facility_roles).any? && resource.admin_editable?
```

Comment:
How about something like `user_has_facility_role?` and `cancer_center_price_group?`?